sdk/monitor/Azure.Monitor.Ingestion/api/Azure.Monitor.Ingestion.net461.cs

```diff
@@ -19,11 +19,20 @@ public enum ServiceVersion
             V2021_11_01_Preview = 1,
         }
     }
+    public partial class UploadFailedArgs : Azure.SyncAsyncEventArgs
+    {
+        internal UploadFailedArgs() : base (default(bool), default(System.Threading.CancellationToken)) { }
+        public System.Exception Exception { get { throw null; } }
+        public int FailedLogs { get { throw null; } }
+    }
     public partial class UploadLogsOptions
     {
         public UploadLogsOptions() { }
         public int MaxConcurrency { get { throw null; } set { } }
         public Azure.Core.Serialization.ObjectSerializer Serializer { get { throw null; } set { } }
+        public event Azure.Core.SyncAsyncEventHandler<Azure.Monitor.Ingestion.UploadFailedArgs> UploadFailed { add { } remove { } }
+        protected internal static void OnException(int logsCount, Azure.Monitor.Ingestion.UploadLogsOptions options, Azure.Response response, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { }
```

Comment:
This should just be `internal`.  There's no need to make it visible to customers with `protected` too.  I'd also give it a more specific name than `OnException` because it's just for raising `UploadFailed`.

Reply:
I thought it needs to be protected for extensibility/testing. At least these are exposed as protected in Search/Service Bus/Event Hubs.

Reply:
Sure, there's a guideline in section 5.4 that says:

> **DO** use a protected virtual method to raise each event. This is applicable only to nonstatic events on unsealed classes—not to structs, sealed classes, or static events.
> 
> For each event, include a corresponding protected virtual method that raises the event. The purpose of the method is to provide a way for a derived class to handle the event using an override. Overriding is a more flexible, faster, and more natural way to handle base class events in derived classes. By convention, the name of the method should start with “On” and be followed with the name of the event.

Feel free to follow that, but it would look a lot different than this `OnException` method.

Reply:
Yeah, agreed that the API would need to change. My suggestion was to have it take a parameter for the UploadFailedEventArgs rather than the current parameters that it uses now.